python/cudf/dataframe/series.py

```diff
@@ -1833,11 +1833,32 @@ def quantile(self, q=0.5, interpolation='linear', exact=True,
         DataFrame
 
         """
+
+        if isinstance(q, (int, float)) and not (0 <= q <= 1):
+            raise ValueError("percentiles should all \
+                             be in the interval [0, 1]")
+
+        if isinstance(q, (int, float)):
+            res = self._column.quantile(q, interpolation, exact)
+            if len(res) == 0:
+                return np.nan
+            else:
+                # if q is an int/float, we shouldn't be constructing series
+                return res.pop()
+
         if not quant_index:
             return Series(self._column.quantile(q, interpolation, exact))
         else:
-            return Series(self._column.quantile(q, interpolation, exact),
-                          index=as_index(np.asarray(q)))
+            from cudf.dataframe.columnops import column_empty_like
+            np_array_q = np.asarray(q)
```

Comment:
After here you should use numpy built-ins to ensure that `0 <= q <= 1`

Reply:
will not need a check here as I'm doing it before this if/else block.